tests/test_upload.py

```diff
@@ -100,3 +104,28 @@ def test_get_config_old_format(tmpdir):
             "more info: "
             "https://docs.python.org/distutils/packageindex.html#pypirc\n"
         ).format(pypirc)
+
+
+def test_skip_existing_skips_files_already_on_PyPI(monkeypatch):
+    response = mock.Mock(spec=upload.requests.models.Response)
+    session = mock.Mock(spec=upload.requests.sessions.Session)
+
+    monkeypatch.setattr(upload.requests, 'session', lambda: session)
+    monkeypatch.setattr(upload.os.path, 'exists', lambda args: True)
+
+    session.post.return_value = response
+    response.status_code = 400
+    response.is_redirect = False
+
+    upload.upload(
+        dists=[WHEEL_FIXTURE],
+        repository="pypi",
+        sign=None,
+        identity=None,
+        username='username',
+        password='password',
+        comment=None,
+        sign_with=None,
+        config_file='',
+        skip_existing=True
+    )
```

Comment:
That no exception is raised is the assertion. If an exception is raised the test will fail/error.